dmcontent/questions.py

```diff
@@ -41,10 +41,26 @@ def filter(self: TQuestion, context, dynamic=True, inplace_allowed: bool = False
 
     def _should_be_shown(self, context):
         return all(
-            depends["on"] in context and (context[depends["on"]] in depends["being"])
+            self._is_dependency_valid(depends["on"], depends["being"], context)
             for depends in self.get("depends", [])
         )
 
+    def _is_dependency_valid(self, key, condition, context):
+        is_valid = False
+        if key in context:
+            if context[key] in condition:
+                is_valid = True
+            else:
+                boolean_array = []
+                for cond in condition:
+                    if "type" in cond and cond["type"] == "integer":
+                        min_value = cond.get("min_value", 0)
+                        max_value = cond.get("max_value", min_value) + 1
+                        boolean_array.append(context[key] in range(min_value, max_value))
```

Comment:
The `+ 1` surprised me - I assume this is for making an inclusive check? Could we use the comparison operators to make this clearer - something like `min <= context[key] <= max`, perhaps?